core/domain/takeout_service.py

```diff
@@ -46,6 +47,32 @@ def get_models_which_should_be_exported():
             base_models.EXPORT_POLICY.CONTAINS_USER_DATA]
 
 
+def export_all_profiles_data_by_parent_user(parent_user_id):
+    """Exports selected models according to model defined export_data functions
+        for all profile users linked to the parent_user of the given id.
+
+    Args:
+        parent_user_id: str. The user_id of the user whose associated profile
+            users' data has being exported.
+
+    Returns:
+        list. List of dictionaries containing all user data in the following
+        format:
+        {
+            <MODEL_NAME>_data: <dict of data in format as specified by
+                                model export policy>
+        }. List might be empty if no profiles are associated with the user
+        corresponding to the given parent_user_id.
+    """
+    all_profiles_user_ids = (
+        user.user_id for user in
+        user_services.get_all_profiles_auth_details_by_parent_user_id(
+            parent_user_id)
+    )
+    return [export_data_for_user(user_id) for user_id in
+            all_profiles_user_ids]
+
+
 def export_data_for_user(user_id):
```

Comment:
Shouldn't we call the above function here? Like when you export data for the user, we should export data of all profiles of the user if the user is a full account right ? (@seanlip to confirm if I am wrong)

Reply:
That's interesting. I'm not sure that's the case actually, if profiles are treated as separate users. This is really more of a question for @BenHenning but if he doesn't get a chance to look at this then I lean towards no. (We can say no for now and table it for discussion in the next Android team meeting perhaps?)

Reply:
@seanlip Can you clarify a bit about what you mean by No. Like what are you leaning towards:
- when we provide a full user's id - all profiles + full user data returned **vs** just full user's data (and call the other function for all profiles also)
- when we provide a profile user's id - return just the profile's data **vs** not allowing this option (prohibiting this from the controllers itself using a proper decorator).
I wanted to know your (or @BenHenning's) preference for each of these options. Thanks!

Reply:
Sure -- I think that when we provide a full user's ID, we export just the full user's data, and when we provide a profile user's ID, we export just the profile's data (or disallow this option until we figure out how exactly profiles interact with the web app).

Reply:
Sounds good , let's keep them separate for now. The change is quite trivial so we can change it later if we decide to do it the way I had thought.

Reply:
Then I think I should keep my current implementation as it is.
- export_data_for_user(full_user_id) - just the full user's data (profile id cannot be provided here because in the controller we have the decorator which doesn't allow profiles to call this).
- export_all_profiles_data_by_parent_user(full_user_id) - get all associated profiles' data.

And not allowing profiles to directly export data for them (as you said disallowing this option until we figure out how exactly profiles interact with the web app).

@seanlip Did I get it correctly? (Sorry but I wanted to sure that I got it properly)

Reply:
I disagree with your second bullet point, I don't think we should have that functionality yet.

Reply:
So basically we want to just export data for full users and don't support for it for profiles.
I think in that case I don't need to make any changes in the takeout_service. (Adding this new function `export_all_profiles_data_by_parent_user` and writing tests for profiles were the only changes I had made).

Reply:
Correct, I think we can remove export_all_profiles_data_by_parent_user. But we should add a test for exporting a profile users data and verify that the correct profile data was exported

Reply:
Yes, tests for that are already there.